src/unix.rs

```diff
@@ -191,14 +196,35 @@ impl Evented for EventedReceiver {
         events: Ready,
         opts: PollOpt,
     ) -> io::Result<()> {
-        let fd = globals().receiver.as_raw_fd();
+        let globals = globals();
+        let fd = globals.receiver.as_raw_fd();
+
+        // NB: we must try registering with the event loop regardless of our
+        // reference count. If the event loop descriptor is closed, we won't
+        // get any notifications to clean up our map, so if a new event loop
+        // descriptor collides with an entry in our map, we want to make sure
+        // we don't accidentally skip the registration.
         match EventedFd(&fd).register(poll, token, events, opts) {
-            Ok(()) => Ok(()),
+            Ok(()) => {}
             // Due to tokio-rs/tokio-core#307
-            Err(ref e) if e.kind() == io::ErrorKind::AlreadyExists => Ok(()),
-            Err(e) => Err(e),
+            Err(ref e) if e.kind() == io::ErrorKind::AlreadyExists => {},
+            Err(e) => return Err(e),
+        }
+
+        let mut guard = globals.pollfd_register_count.lock().expect("poisoned");
+        match guard.entry(poll.as_raw_fd()) {
+            Entry::Occupied(mut occupied) => {
+                let mut val = occupied.get_mut();
+                *val = val.saturating_add(1);
+            },
+            Entry::Vacant(vacant) => {
+                vacant.insert(1);
```

Comment:
I think it's safe to say that it's impossible for this to overflow, so this can probably just be:

```rust
*globals.pollfd_register_count
    .lock()
    .expect("poisoned")
    .entry(poll.as_raw_fd())
    .or_insert(0) += 1;
```

(similarly below)